pandas/_libs/tslibs/_timestamp.pyx

```diff
@@ -0,0 +1,420 @@
+                raise ValueError("Cannot add integral value to Timestamp "
+                                 "without freq.")
+            return self.__class__((self.freq * other).apply(self),
+                                  freq=self.freq)
+
+        elif PyDelta_Check(other) or hasattr(other, 'delta'):
+            # delta --> offsets.Tick
+            # logic copied from delta_to_nanoseconds to prevent circular import
+            if hasattr(other, 'nanos'):
+                nanos = other.nanos
+            elif hasattr(other, 'delta'):
+                nanos = other.delta
+            elif PyDelta_Check(other):
+                nanos = (other.days * 24 * 60 * 60 * 1000000 +
+                         other.seconds * 1000000 +
+                         other.microseconds) * 1000
+
+            result = self.__class__(self.value + nanos,
+                                    tz=self.tzinfo, freq=self.freq)
+            if getattr(other, 'normalize', False):
+                # DateOffset
+                result = result.normalize()
+            return result
+
+        # index/series like
+        elif hasattr(other, '_typ'):
+            return NotImplemented
+
+        result = datetime.__add__(self, other)
+        if PyDateTime_Check(result):
+            result = self.__class__(result)
+            result.nanosecond = self.nanosecond
+        return result
+
+    def __sub__(self, other):
+        if (is_timedelta64_object(other) or is_integer_object(other) or
+                PyDelta_Check(other) or hasattr(other, 'delta')):
+            # `delta` attribute is for offsets.Tick or offsets.Week obj
+            neg_other = -other
+            return self + neg_other
+
+        typ = getattr(other, '_typ', None)
+
+        # a Timestamp-DatetimeIndex -> yields a negative TimedeltaIndex
+        if typ in ('datetimeindex', 'datetimearray'):
+            # timezone comparison is performed in DatetimeIndex._sub_datelike
+            return -other.__sub__(self)
+
+        # a Timestamp-TimedeltaIndex -> yields a negative TimedeltaIndex
+        elif typ in ('timedeltaindex', 'timedeltaarray'):
+            return (-other).__add__(self)
+
+        elif other is NaT:
+            return NaT
+
+        # coerce if necessary if we are a Timestamp-like
+        if (PyDateTime_Check(self)
+                and (PyDateTime_Check(other) or is_datetime64_object(other))):
+            if isinstance(self, _Timestamp):
+                other = self.__class__(other)
+            else:
+                self = other.__class__(self)
+
+            # validate tz's
+            if not tz_compare(self.tzinfo, other.tzinfo):
+                raise TypeError("Timestamp subtraction must have the "
+                                "same timezones or no timezones")
+
+            # scalar Timestamp/datetime - Timestamp/datetime -> yields a
+            # Timedelta
+            try:
+                return self._create_timedelta(self.value - other.value)
```

Comment:
`Timedelta` -> `self._create_timedelta`

`_create_timedelta` is a new helper function in `Timestamp` to create `Timedelta`s.  Not the most elegant solution, but didn't have a good workaround here.

Reply:
just do a runtime import of Timedelta

Reply:
I was hoping to avoid that, but it has been done